impl_core/src/main/java/io/opencensus/implcore/trace/export/SpanExporterImpl.java

```diff
@@ -187,27 +177,43 @@ public void run() {
               }
             } while (spans.isEmpty());
           }
-          spansCopy = new ArrayList<RecordEventsSpanImpl>(spans);
+          spansCopy = new ArrayList<>(spans);
           spans.clear();
         }
         // Execute the batch export outside the synchronized to not block all producers.
-        final List<SpanData> spanDataList = fromSpanImplToSpanData(spansCopy);
-        if (!spanDataList.isEmpty()) {
-          onBatchExport(spanDataList);
-        }
+        exportBatches(spansCopy);
       }
     }
 
-    void flush() {
-      List<RecordEventsSpanImpl> spansCopy;
+    private void flush() {
+      ArrayList<RecordEventsSpanImpl> spansCopy;
       synchronized (monitor) {
-        spansCopy = new ArrayList<RecordEventsSpanImpl>(spans);
+        spansCopy = new ArrayList<>(spans);
         spans.clear();
       }
+      // Execute the batch export outside the synchronized to not block all producers.
+      exportBatches(spansCopy);
+    }
 
-      final List<SpanData> spanDataList = fromSpanImplToSpanData(spansCopy);
+    private void exportBatches(ArrayList<RecordEventsSpanImpl> spanList) {
+      ArrayList<SpanData> spanDataList = new ArrayList<>(bufferSize);
```

Comment:
Why not just use `Lists.partition`?

Reply:
Never mind, forgot we don't want to depend on Guava here.

Reply:
Not only that, but we want to remove the reference to the RecordEventsSpanImpl asap, with Lists.partitions they use List.subList which is implemented as a wrapper on top of the original list also immutable so it is harder to remove the reference to that object.